app/component/qrcode.tsx

```diff
@@ -0,0 +1,61 @@
+import QRCodeStyling, { type Options } from "qr-code-styling"
+import { useEffect, useRef, type FC } from "react"
+
+import byte from "~packages/util/byte"
+import type { HexString } from "~typing"
+
+interface QrCodeProps {
+  address: HexString
+}
+
+export const QrCode: FC<QrCodeProps> = ({ address }) => {
+  const qrRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    // Refer to:
+    // https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/ui/qr-code-view/qr-code-view.tsx#L36-L38
+    const data = `${byte.isHex(address) ? "ethereum:" : ""}${address}`
+
+    // Refer to:
+    // https://github.com/kozakdenys/qr-code-styling-examples/blob/master/examples/react/src/App.tsx
+    const option: Options = {
+      type: "svg",
+      shape: "square",
+      width: 185,
+      height: 185,
+      data,
+      qrOptions: {
+        errorCorrectionLevel: "Q"
+      },
+      dotsOptions: {
+        color: "#000000",
+        type: "dots"
+      },
+      backgroundOptions: {
+        color: "#ffffff"
+      },
+      cornersSquareOptions: {
+        color: "#000000",
+        type: "extra-rounded"
+      },
+      cornersDotOptions: {
+        color: "#000000",
+        type: "square"
+      }
+    }
+    const qrCode = new QRCodeStyling(option)
+    if (qrRef.current) {
+      qrCode.append(qrRef.current)
+    }
+    if (qrCode) {
+      qrCode.update(option)
+    }
```

Comment:
Maybe we could remove `if (qrCode)`.

Reply:
Yes, in normal use, the `if (qrCode)` can be removed.
After testing, if the user changes the address accidentally, the QR code will be `appended` as follows:

<img width="401" alt="截圖 2024-07-19 下午3 30 43" src="https://github.com/user-attachments/assets/1ba8299f-3cb3-477a-b7bf-ee2a6a079dc2">

Reply:
I see. 🧐 I'm curious how to reproduce this.

Reply:
You can follow these steps when running `npm run dev` to reproduce this:

1. Remove `if (qrCode){ ... }`.
2. Save the file, and it will auto recompile to re-append the QR code.